test/test_utils/__init__.py

```diff
@@ -1510,6 +1510,35 @@ def get_python_version_from_image_uri(image_uri):
     python_version = python_version_search.group()
     return "py36" if python_version == "py3" else python_version
 
+def construct_buildspec_path(dlc_path, framework_path, buildspec, framework_version):
+    """
+    Construct a relative path to the buildspec yaml file by iterative checking on the existence of
+    a specific version file for the framework being tested. Possible options include:
+    [buildspec-[Major]-[Minor]-[Patch].yml, buildspec-[Major]-[Minor].yml, buildspec-[Major].yml, buildspec.yml]
+    :param dlc_path: path to the DLC test folder
+    :param framework_path: Framework folder name
+    :param buildspec: buildspec file name
+    :param framework_version: default (long) framework version name
+    """
+    if framework_version:
```

Comment:
Yes. The default path can be built without using the framework version. See last line of function.

Reply:
Framework version is derived from image uri. In the case where framework version cannot be retrieved from image, the framework version can be empty. This is very unlikely given the naming convention used for DLC images. See function get_framework_and_version_from_tag for how to get version of an image uri. I don't think we need to provide a default value since we expect that framework version will always be provided.

Reply:
Sure, we can look at this as addressing an edge case